keps/sig-storage/4049-storage-capacity-scoring-of-nodes-for-dynamic-provisioning/README.md

```diff
@@ -0,0 +1,1021 @@
+
+<!--
+**Note:** This checklist is iterative and should be reviewed and updated every time this enhancement is being considered for a milestone.
+-->
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+This KEP proposes adding a way to score nodes for dynamic provisioning of PVs. This scoring method is based on storage capacity in the VolumeBinding plugin. 
+By considering the amount of free space that nodes have, it is possible to dynamically schedule pods on the node that has the most or least free space.
+
+<!--
+This section is incredibly important for producing high-quality, user-focused
+documentation such as release notes or a development roadmap. It should be
+possible to collect this information before implementation begins, in order to
+avoid requiring implementors to split their attention between writing release
+notes and implementing the feature itself. KEP editors and SIG Docs
+should help to ensure that the tone and content of the `Summary` section is
+useful for a wide audience.
+
+A good summary is probably at least a paragraph in length.
+
+Both in this section and below, follow the guidelines of the [documentation
+style guide]. In particular, wrap lines to a reasonable length, to make it
+easier for reviewers to cite specific portions, and to minimize diff churn on
+updates.
+
+[documentation style guide]: https://github.com/kubernetes/community/blob/master/contributors/guide/style-guide.md
+-->
+
+## Motivation
+
+Storage capacity needs to be considered when:
+
+- we want to resize after a node-local PV is scheduled. In this case we need to select a node with as much free space as possible. 
+- we want to select a node with less free node space to reduce the number of nodes as much as possible.
+
+<!--
+This section is for explicitly listing the motivation, goals, and non-goals of
+this KEP.  Describe why the change is important and the benefits to users. The
+motivation section can optionally provide links to [experience reports] to
+demonstrate the interest in a KEP within the wider Kubernetes community.
+
+[experience reports]: https://github.com/golang/go/wiki/ExperienceReports
+-->
+
+### Goals
+
+- To modify the scoring logic to count on dynamic provisioning in addition to the current, considering only static provisioning.
+
+<!--
+List the specific goals of the KEP. What is it trying to achieve? How will we
+know that this has succeeded?
+-->
+
+### Non-Goals
+
+- To change how to score nodes for static provisioning.
+
+<!--
+What is out of scope for this KEP? Listing non-goals helps to focus discussion
+and make progress.
+-->
+
+## Proposal
+
+- Node scores based on available space can be taken into account when performing dynamic provisioning.
```

Comment:
Please add a note *what* is the proposed configuration mechanism, like:

> Cluster admin can configure the scoring logic using a new field in `kubescheduler.config.k8s.io` `VolumeBindingargs` (with a link to `types_pluginargs.go`). The scoring logic is global for the whole cluster and we propose two values:
> - Prefer a node with the least capacity.
> - Prefer a node with the maximum capacity.

It is buried in the detailed design below and hard to spot there.

Also, here and perhaps in the details below, we should specify what happens when no policy is chosen (which is going to be the default), e.g:

> When unset, no scoring logic is used. Pods will be placed to any node that has enough capacity.

Reply:
> Please add a note what is the proposed configuration mechanism, like:

Thank you. I added it.

> Also, here and perhaps in the details below, we should specify what happens when no policy is chosen (which is going to be the default), e.g:

I have added an explanation of the default behavior here and in the "Design Details." The default behavior is to prefer a node with the maximum capacity. This was originally written in the "User Stories," but I have removed it from there.